darshan-util/pydarshan/darshan/experimental/plots/plot_dxt_heatmap.py

```diff
@@ -397,14 +398,14 @@ def plot_heatmap(
     colorbar_kws = {"label": colorbar_label}
     # create the heatmap object using the heatmap data,
     # and assign it to the jointplot main figure
-    hmap = sns.heatmap(
-        hmap_df,
-        ax=jgrid.ax_joint,
-        # choose a color map that is not white at any value
-        cmap="YlOrRd",
-        norm=LogNorm(),
-        cbar_kws=colorbar_kws,
-    )
+
+    x, y = np.meshgrid(np.arange(xbins),
+                       np.asarray(hmap_df.index))
+    # x and y both have shape (active_ranks, xbins)
+    # rather than (nprocs, xbins)
+    hmap = jgrid.ax_joint.scatter(x, y, c=hmap_df, cmap="YlOrRd", norm=LogNorm(), marker="s")
```

Comment:
Either for my own reference, or if i.e., a student comes back to look at this later--one idea might be a custom rectangle/marker if we could control the dimensions with sufficient granularity. I believe that since the number of ranks and bin widths are fixed for a given plot, we should only have to do the calculation once and then be able to reuse that for each "scatter marker."

Maybe something similar to: https://stackoverflow.com/a/58552620/2942522

Reply:
If we can sort that out, I think it is worth it, since we potentially eliminate thousands of `float64` `0`s, in case that is not clear.